fs/xfstests.py

```diff
@@ -84,42 +96,90 @@ def setUp(self):
                       ", ".join(self.available_tests))
         process.run('useradd fsgqa', sudo=True)
         process.run('useradd 123456-fsgqa', sudo=True)
+        self.scratch_mnt = '/mnt/scratch'
+        self.test_mnt = '/mnt/test'
+        if not os.path.exists(self.scratch_mnt):
+            os.makedirs(self.scratch_mnt)
+        if not os.path.exists(self.test_mnt):
+            os.makedirs(self.test_mnt)
 
     def test(self):
         failures = False
         os.chdir(self.srcdir)
-        for test in self.test_list:
-            test = '*/' + test
-            cmd = './check %s' % test
-            result = process.run(cmd, ignore_status=True)
+        if not self.test_list:
+            self.log.info('Running all tests')
+            cmd = './check -g auto'
+            result = process.run(cmd, ignore_status=True, verbose=True)
             if result.exit_status == 0:
-                self.log.info('OK: Test %s passed.', test)
+                self.log.info('OK: All Tests passed.')
             else:
                 msg = self._parse_error_message(result.stdout)
-                self.log.info('ERR: %s failed. Message: %s', test, msg)
+                self.log.info('ERR: Test(s) failed. Message: %s', msg)
                 failures = True
+
+        else:
+            self.log.info('Running only specified tests')
+            for test in self.test_list:
+                test = '%s/%s' % (self.fs_to_test, test)
+                cmd = './check %s' % test
+                result = process.run(cmd, ignore_status=True, verbose=True)
+                if result.exit_status == 0:
+                    self.log.info('OK: Test %s passed.', test)
+                else:
+                    msg = self._parse_error_message(result.stdout)
+                    self.log.info('ERR: %s failed. Message: %s', test, msg)
+                    failures = True
         if failures:
             self.fail('One or more tests failed. Please check the logs.')
 
     def tearDown(self):
         process.system('userdel fsgqa', sudo=True)
         process.system('userdel 123456-fsgqa', sudo=True)
+        # In case if any test has been interrupted
+        process.system('umount self.scratch_mnt self.test_mnt',
+                       sudo=True, ignore_status=True)
+        if os.path.exists(self.scratch_mnt):
+            os.rmdir(self.scratch_mnt)
+        if os.path.exists(self.test_mnt):
+            os.rmdir(self.test_mnt)
+        if self.dev_type == 'loop':
+            process.system(
+                'for i in $(seq 0 1); do losetup -d '
+                '/dev/loop$i; done', shell=True, sudo=True)
```

Comment:
How are you doing handle error condition over here (if `/dev/loop0, /dev/loop1` is not availble)